### DISCUSSION TITLE
TLS? Surely there's an easier way!

### DISCUSSION THREAD

**Discussion by DirectITServicesUK** [postalserver/postal#2177] [Help with using Postal]
So, Postal installed. But it doesn't send / recieve emails using certificates by default. Surely in this day and age that part of the set up would be included in the initial set up. I've spent 2 days hacking away, and searching for the right answer, but just can't figure the process out. I'm not a noob.

What IS the correct way to enable TLS and get the certs in the right place? Ideally auto-updating certs.

**Comment by willpower232** [comment]
You can provide a certificate you have purchased or generated with Lets Encrypt to Postal via the config file

https://github.com/postalserver/postal/blob/83fef0e8a01285e6c9cb839d72d2cfca2371c839/config/postal.defaults.yml#L68-L70

You will probably need to restart Postal when the certificate updates.

I'll make a note that this is missing from the documentation.

**Comment by DirectITServicesUK** [comment]
I've made a script now that shuts down caddy, sorts the certs, check "ln" to the configuration directory and restarts caddy.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

DOUG HALL
MANAGING DIRECTOR
Direct IT Services Ltd
Mills House, Boscombe Down Business Park
Amesbury
SP4 7RX
Tel: 01980 348271
Mobile: 07597 194452
Email: ***@***.***
Web: directitservices.co.uk
IT Support portal - https://support.itportal.uk
 Direct IT Services Ltd accepts no liability for the content of this email, or for the consequences of any actions taken on the basis of the information provided, unless that information is subsequently confirmed in writing. Any views or opinions presented in this email are solely those of the author and do not necessarily represent those of the company.
 
 Direct IT Services Ltd is a company registered in England and Wales under number 13656277.
 Registered office: Mills House Mills Way, Boscombe Down Business Park, Amesbury, Wiltshire, England, SP4 7RX

**Reply by willpower232** [reply]
I thought that caddy sorted out its own certificates so this sounds a bit more overkill than most would need but as long as you eventually restart Postal if you give it certificates, that should be fine :sweat_smile:
